src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx

```diff
@@ -98,7 +114,60 @@ export default function CampaignFormModal({isOpen, handleClose, apiSettings, tit
         },
     });
 
-    const onSubmit: SubmitHandler<Inputs> = async (inputs) => {
+    const image = watch('image');
+    const selectedGoalType = watch('goalType');
+    const goal = watch('goal');
+
+    useEffect(() => {
+        switch (step) {
+            case 1:
+                setFormModalTitle(__('Tell us about your fundraising cause', 'give'));
+                break;
+            case 2:
+                setFormModalTitle(__('Set up your campaign goal', 'give'));
+                break;
+        }
+    }, [step]);
+
+    useEffect(() => {
+        switch (selectedGoalType) {
+            case 'amount':
+                setGoalInputAttributes({
```

Comment:
Is configuring dynamic values using state necessary with the form library?  We might want to consider a lookup variable instead of a state change within a `useEffect` hook if possible. In particular, I think changing state within a `useEffect`, which is triggered by a state change can have unintended side effects.

Suggestion:
```tsx
const goalInputAttributes = {
    amount: {
        label: 'How much do you want to raise?'
    },
    donations: {
        label: 'How much do you want to raise?'
    }
}


goalInputAttributes[selectedGoalType).label
```

Reply:
Resolved in 2987ed6171f678e6c0b7da3859715891e7e63072